csharp/source/autowrap/csharp/csharp.d

```diff
@@ -0,0 +1,1061 @@
+module autowrap.csharp.csharp;
+
+import autowrap.csharp.boilerplate;
+import autowrap.csharp.common;
+import autowrap.reflection;
+
+import std.ascii;
+import std.datetime;
+import std.traits;
+import std.string;
+import std.uuid;
+
+import std.stdio;
+
+private __gshared string[string] returnTypes;
```

Comment:
Except in the case of interfacing with C code, `__gshared` is almost always a mistake, and when it's not, it should be accompanied by a comment explaining why it's being used.